docker/recorder/src/main/java/org/arquillian/cube/docker/impl/client/reporter/TakeDockerEnvironment.java

```diff
@@ -189,8 +276,136 @@ private File createSchemasDirectory(File rootDirectory) {
         return schemasDir.toFile();
     }
 
+    private File createContainerLogDirectory(File rootDirectory) {
+        final Path reportsLogs = Paths.get("reports", "logs");
+        final Path logsDir = rootDirectory.toPath().resolve(reportsLogs);
+        if (Files.notExists(logsDir)) {
+            try {
+
+                Files.createDirectories(logsDir);
+            } catch (IOException e) {
+                throw new IllegalArgumentException(String.format("Could not created logs directory at %s", logsDir));
+            }
+        }
+        return logsDir.toFile();
+    }
+
     private void addEntry(PropertyEntry propertyEntry, GroupEntry groupEntry) {
         groupEntry.getPropertyEntries().add(propertyEntry);
     }
 
+    private Map<String, Map<String, ?>> getStats(Statistics statistics, Boolean decimal) {
+
+        Map<String, Map<String, ?>> stats = new LinkedHashMap<>();
+        if (statistics != null){
+            Map<String, Map<String, String>> networks = extractNetworksStats(statistics.getNetworks(), decimal);
+            Map<String, String> memory = extractMemoryStats(statistics.getMemoryStats(), decimal, "usage", "max_usage", "limit");
+            Map<String, String> blkio = extractIORW(statistics.getBlkioStats(), decimal);
+            stats.put("network", networks);
+            stats.put("memory", memory);
+            stats.put("block I/O", blkio);
+        }
+        return stats;
+    }
+
+    private Map<String, Map<String, String>> extractNetworksStats(Map<String, Object> map, boolean decimal) {
```

Comment:
This class has at least 3 responsibilities now

- listens to the Cube events and dispatches some logic
- extract docker-specific information
- does number conversion

I would suggest to split it to three components and do the same for unit tests.

WDYT @lordofthejars @dipak-pawar